pkg/admission/validator/webhook.go

```diff
@@ -43,3 +46,17 @@ func New(mgr manager.Manager) (*extensionswebhook.Webhook, error) {
 		},
 	})
 }
+
+// NewSecretsWebhook creates a new validation webhook for Secrets.
+func NewSecretsWebhook(mgr manager.Manager) (*extensionswebhook.Webhook, error) {
```

Comment:
Was it necessary to create a new endpoint or why didn't you just add the `NewSecretValidator()` to the existing one?

Reply:
Yep, generally it would be good to have only 1 endpoint and I recall I had difficulties naming the new one. The issue currently is that the existing webhook handler specifies 

```
		Predicates: []predicate.Predicate{extensionspredicate.GardenCoreProviderType(gcp.Type)},
```

which is perfectly fine for the Shoot and CloudProfile but it will return always `false` for `corev1.Secret`.

Reply:
Oh, I see. Thanks for explaining!